homeassistant/components/sensor/forecast.py

```diff
@@ -124,80 +116,66 @@ def state(self):
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
-        return self._unit_of_measurement
+        self.forecast_data.update_unit_of_measurement(self.type)
```

Comment:
Will the unit of measurement ever change during the lifetime of this object?
You also probably need to lock this in some way if you do it this way, since another property updating in another thread can also call this and the forecasr_data.unit of measurement will be changed to that type